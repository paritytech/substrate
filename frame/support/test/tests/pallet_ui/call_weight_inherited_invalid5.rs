// Stray tokens after good input.

#[frame_support::pallet]
mod parentheses {
	use super::*;
	
	#[pallet::config]
	pub trait Config: frame_system::Config {
	}

	#[pallet::pallet]
	pub struct Pallet<T>(core::marker::PhantomData<T>);

	#[pallet::call(weight(<T as Config>::WeightInfo straycat))]
	impl<T: Config> Pallet<T> {
		#[pallet::call_index(0)]
		pub fn foo(_: OriginFor<T>) -> DispatchResult {
			Ok(())
		}
	}
}

#[frame_support::pallet]
mod assign {
	use super::*;
	
	#[pallet::config]
	pub trait Config: frame_system::Config {
	}

	#[pallet::pallet]
	pub struct Pallet<T>(core::marker::PhantomData<T>);

	#[pallet::call(weight = <T as Config>::WeightInfo straycat)]
	impl<T: Config> Pallet<T> {
		#[pallet::call_index(0)]
		pub fn foo(_: OriginFor<T>) -> DispatchResult {
			Ok(())
		}
	}
}

fn main() {
}
