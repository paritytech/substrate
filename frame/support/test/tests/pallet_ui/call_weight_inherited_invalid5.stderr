error: unexpected token
  --> tests/pallet_ui/call_weight_inherited_invalid5.rs:14:50
   |
14 |     #[pallet::call(weight(<T as Config>::WeightInfo straycat))]
   |                                                     ^^^^^^^^

error: unexpected token
  --> tests/pallet_ui/call_weight_inherited_invalid5.rs:34:52
   |
34 |     #[pallet::call(weight = <T as Config>::WeightInfo straycat)]
   |                                                       ^^^^^^^^
